kratos/tests/cpp_tests/includes/test_property_accessor.cpp

```diff
@@ -0,0 +1,95 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "testing/testing.h"
+#include "includes/properties.h"
+#include "geometries/quadrilateral_2d_4.h"
+#include "tests/cpp_tests/auxiliar_files_for_cpp_unnitest/test_element.h"
+#include "tests/cpp_tests/auxiliar_files_for_cpp_unnitest/test_constitutive_law.h"
+
+namespace Kratos::Testing 
+{
+using NodeType = Node<3>;
+
+/**
+* Checks the correct work of the Has methods
+*/
+KRATOS_TEST_CASE_IN_SUITE(PropertyAccessorSimpleProperties, KratosCoreFastSuite)
+{
+    Model current_model;
+    auto &this_model_part = current_model.CreateModelPart("ModelPart",1);
+    auto p_prop = this_model_part.CreateNewProperties(0);
+    p_prop->SetValue(YOUNG_MODULUS, 2.1e11);
+
+    const double initial_E = ((*p_prop)[YOUNG_MODULUS]);
+    KRATOS_CHECK_LESS_EQUAL(2.1e11 - initial_E, 1.0e-8);
+
+    // custom accessor that returns 2.0
+    class CustomAccessor
+        : public  Accessor 
+    {
+        public:
+        double GetValue(
+            const Variable<double> &rVariable,
+            const Properties &rProperties,
+            const GeometryType &rGeometry,
+            const Vector &rShapeFunctionVector,
+            const ProcessInfo &rProcessInfo) const override
+        {
+            return mValue * rProperties[rVariable];
+        }
+
+        Accessor::Pointer Clone() const override
+        {
+            return Kratos::make_shared<CustomAccessor>(*this);
+        }
+
+        private:
+            const double mValue = 2.0;
+    };
+
+    auto& r_process_info = this_model_part.GetProcessInfo();
+    TestConstitutiveLaw r_clone_cl = TestConstitutiveLaw();
+    p_prop->SetValue(CONSTITUTIVE_LAW, r_clone_cl.Clone());
```

Comment:
Do we have to set the `ConstitutiveLaw` in here? Wouldn't the test be better without it?